libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1452,13 +1458,40 @@ async fn pull_transactions(
         .map(TryInto::try_into)
         .collect();
 
+    let outbound_transactions: NodeResult<Vec<Payment>> =
+        htlc_expiry(outbound_transactions.unwrap(), htlc_list);
+
     let mut transactions: Vec<Payment> = Vec::new();
     transactions.extend(received_transactions?);
     transactions.extend(outbound_transactions?);
 
     Ok(transactions)
 }
 
+fn htlc_expiry(payments: Vec<Payment>, htlc_list: Vec<Htlc>) -> NodeResult<Vec<Payment>> {
```

Comment:
The fn name doesn't really tell me what the function does, I had to look at the code. Maybe rename to `update_payment_expirations`